orchagent/notifications.cpp

```diff
@@ -23,12 +23,12 @@ void on_bfd_session_state_change(uint32_t count, sai_bfd_session_state_notificat
     // which causes concurrency access to the DB
 }
 
-void on_switch_shutdown_request()
+void on_switch_shutdown_request(sai_object_id_t switch_id)
 {
     SWSS_LOG_ENTER();
 
     /* TODO: Later a better restart story will be told here */
     SWSS_LOG_ERROR("Syncd stopped");
 
-    exit(EXIT_FAILURE);
+    _exit(EXIT_FAILURE);
```

Comment:
_exit() will exit current process immediately:
https://man7.org/linux/man-pages/man2/exit.2.html

I manually validate this change can fix the issue with following code:


std::shared_ptr<swss::DBConnector>          g_db;
std::shared_ptr<swss::RedisPipeline>        g_redisPipeline;

static void exitHandler()
{
    sleep(2);
    //exit(0);
    _exit(0);
}

static void routeCreater()
{
    while (true)
    {
        cout << "Start redis command" << endl;
        RedisCommand command;
        command.format("HLEN %s", "TEST_KEY");
        g_redisPipeline->push(command, REDIS_REPLY_INTEGER);
        cout << "End redis command" << endl;
    }
}

TEST(RedisPipeline, test)
{
    g_db                 = std::make_shared<swss::DBConnector>(ASIC_DB, swss::DBConnector::DEFAULT_UNIXSOCKET, 0);
    g_redisPipeline      = std::make_shared<swss::RedisPipeline>(g_db.get());

    thread *exitThread = new thread(exitHandler);
    thread *routeCreaterThread = new thread(routeCreater);

    sleep(5);

    exitThread->join();
    routeCreaterThread->join();
    delete exitThread;
    delete routeCreaterThread;
}

Reply:
Check if we can ignore flush in Redispipeline dtor.